specification/datalake-store/resource-manager/Microsoft.DataLakeStore/stable/2016-11-01/account.json

```diff
@@ -1910,6 +2199,85 @@
       },
       "description": "The firewall rule properties to use when updating a firewall rule."
     },
+    "CreateOrUpdateVirtualNetworkRuleParameters": {
+      "required": [
+        "properties"
+      ],
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/CreateOrUpdateVirtualNetworkRuleProperties",
+          "description": "The virtual network rule properties to use when creating a new virtual network rule."
+        }
+      },
+      "description": "The parameters used to create a new virtual network rule."
+    },
+    "CreateVirtualNetworkRuleWithAccountParameters": {
+      "required": [
+        "name",
+        "properties"
+      ],
+      "properties": {
+        "name": {
```

Comment:
name MUST come from the URL. It should not be present in the request body. In request body, it should be marked s readonly.

Reply:
Looks like on the PUT of account, you are providing the option of creating a virtualNetworkRule. This is not supported. VirtualNetworkRule must be explicitly created by doing a PUT on this type. If needed, It can be referenced in the parent resource using the resourceId.